Running RC9 gradlew detekt crashes with detekt-formatting included

## Expected Behavior
Detekt runs fine using Gradle and reports formatter problems

## Current Behavior
Detekt crashes if detekt-formatter is included in Gradle configuration
```
Task :detekt FAILED
Error: Could not find or load main class io.gitlab.arturbosch.detekt.cli.Main

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detekt'.
> Process 'command 'C:\java\jdk1.8.0_131\bin\java.exe'' finished with non-zero exit value 1
```
## Steps to Reproduce (for bugs)
1. Configure Detekt RC9 for Gradle, detektVersion = 1.0.0.RC9
```
buildscript {
       dependencies {
        classpath "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:${detektVersion}"
...
```
2. Using settings:
```
detekt {
    toolVersion = "${detektVersion}"
    input = files("src/main/kotlin","src/test/kotlin")
    config = files("detekt.yml")
    filters = ".*/resources/.*,.*/tmp/.*"
}
```
3. Include detekt-formatter RC9
```
dependencies {
    detekt("io.gitlab.arturbosch.detekt:detekt-formatting:${detektVersion}")
...
```
4. Run gradlew detekt -> detekt crashes
5. Remove detekt-formatting dependency in Gradle configuration and gradlew detekt runs fine

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used: Detekt RC9
* Operating System and version: Win10 & Gradle 4.9
* Link to your project:

**`./gradlew app:detekt --stacktrace`**

```
> Task :app:detekt FAILED
Error: Could not find or load main class io.gitlab.arturbosch.detekt.cli.Main

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:detekt'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_66.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:detekt'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipCachedTaskExecuter.execute(SkipCachedTaskExecuter.java:105)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ResolveBuildCacheKeyExecuter.execute(ResolveBuildCacheKeyExecuter.java:79)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
        at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:277)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:262)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.process.internal.ExecException: Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_66.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1
        at org.gradle.process.internal.DefaultExecHandle$ExecResultImpl.assertNormalExitValue(DefaultExecHandle.java:395)
        at org.gradle.process.internal.DefaultJavaExecAction.execute(DefaultJavaExecAction.java:37)
        at org.gradle.api.internal.file.DefaultFileOperations.javaexec(DefaultFileOperations.java:225)
        at org.gradle.api.internal.project.DefaultProject.javaexec(DefaultProject.java:1102)
        at io.gitlab.arturbosch.detekt.invoke.DetektInvoker.invokeCli$detekt_gradle_plugin(DetektInvoker.kt:13)
        at io.gitlab.arturbosch.detekt.Detekt.check(Detekt.kt:110)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:801)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:768)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        ... 33 more
```

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt
@@ -82,10 +82,10 @@ class DetektPlugin : Plugin<Project> {
 	private fun determineInput(extension: DetektExtension) = extension.input.filter { it.exists() }
 
 	private fun configurePluginDependencies(project: Project, extension: DetektExtension) {
-		project.configurations.create(CONFIGURATION_DETEKT_PLUGIN) {
+		project.configurations.create(CONFIGURATION_DETEKT_INTERNAL) {
 			isVisible = false
 			isTransitive = true
-			description = "The internal $CONFIGURATION_DETEKT_PLUGIN libraries to be used for this project."
+			description = "The internal $CONFIGURATION_DETEKT_INTERNAL libraries to be used for this project."
 
 			@Suppress("USELESS_ELVIS")
 			val version = extension.toolVersion ?: DEFAULT_DETEKT_VERSION
@@ -111,4 +111,4 @@ class DetektPlugin : Plugin<Project> {
 }
 
 const val CONFIGURATION_DETEKT = "detekt"
-const val CONFIGURATION_DETEKT_PLUGIN = "detektPlugins"
+const val CONFIGURATION_DETEKT_INTERNAL = "detektInternal"

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt
@@ -81,17 +81,20 @@ class DetektPlugin : Plugin<Project> {
 
 	private fun determineInput(extension: DetektExtension) = extension.input.filter { it.exists() }
 
-	private fun configurePluginDependencies(project: Project, extension: DetektExtension) =
-			project.configurations.create(DETEKT) {
-				isVisible = false
-				isTransitive = true
-				description = "The $DETEKT libraries to be used for this project."
-				defaultDependencies {
-					@Suppress("USELESS_ELVIS")
-					val version = extension.toolVersion ?: DEFAULT_DETEKT_VERSION
-					add(project.dependencies.create("io.gitlab.arturbosch.detekt:detekt-cli:$version"))
-				}
-			}
+	private fun configurePluginDependencies(project: Project, extension: DetektExtension) {
+		project.configurations.create(DETEKT) {
+			isVisible = false
+			isTransitive = true
+			description = "The $DETEKT libraries to be used for this project."
+
+			@Suppress("USELESS_ELVIS")
+			val version = extension.toolVersion ?: DEFAULT_DETEKT_VERSION
+			dependencies.add(project.dependencies.add(
+					"detekt",
+					"io.gitlab.arturbosch.detekt:detekt-cli:$version")
+			)
+		}
+	}
 
 
 	companion object {

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt
@@ -82,20 +82,24 @@ class DetektPlugin : Plugin<Project> {
 	private fun determineInput(extension: DetektExtension) = extension.input.filter { it.exists() }
 
 	private fun configurePluginDependencies(project: Project, extension: DetektExtension) {
-		project.configurations.create(DETEKT) {
+		project.configurations.create(CONFIGURATION_DETEKT_PLUGIN) {
 			isVisible = false
 			isTransitive = true
-			description = "The $DETEKT libraries to be used for this project."
+			description = "The internal $CONFIGURATION_DETEKT_PLUGIN libraries to be used for this project."
 
 			@Suppress("USELESS_ELVIS")
 			val version = extension.toolVersion ?: DEFAULT_DETEKT_VERSION
-			dependencies.add(project.dependencies.add(
-					"detekt",
-					"io.gitlab.arturbosch.detekt:detekt-cli:$version")
-			)
+			defaultDependencies {
+				add(project.dependencies.create("io.gitlab.arturbosch.detekt:detekt-cli:$version"))
+			}
+		}
+
+		project.configurations.create(CONFIGURATION_DETEKT) {
+			isVisible = false
+			isTransitive = true
+			description = "The $CONFIGURATION_DETEKT libraries to be used for this project."
 		}
 	}
-
 
 	companion object {
 		private const val DETEKT = "detekt"
@@ -105,3 +109,6 @@ class DetektPlugin : Plugin<Project> {
 		private const val BASELINE = "detektBaseline"
 	}
 }
+
+const val CONFIGURATION_DETEKT = "detekt"
+const val CONFIGURATION_DETEKT_PLUGIN = "detektPlugins"

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
@@ -1,19 +1,33 @@
 package io.gitlab.arturbosch.detekt.invoke
 
+import io.gitlab.arturbosch.detekt.CONFIGURATION_DETEKT
+import io.gitlab.arturbosch.detekt.CONFIGURATION_DETEKT_PLUGIN
 import org.gradle.api.Project
+import org.gradle.api.file.FileCollection
 
 /**
  * @author Marvin Ramin
  */
 object DetektInvoker {
 	internal fun invokeCli(project: Project, arguments: List<CliArgument>, debug: Boolean = false) {
-		val args = arguments.map(CliArgument::toArgument).flatten()
+		val cliArguments = arguments.map(CliArgument::toArgument).flatten()
 
-		if (debug) println(args)
+		if (debug) println(cliArguments)
 		project.javaexec {
-			main = "io.gitlab.arturbosch.detekt.cli.Main"
-			classpath(project.configurations.getAt("detekt"))
-			args(args)
+			main = DETEKT_MAIN
+			classpath = getConfigurations(project, debug)
+			args = cliArguments
 		}
 	}
+
+	private fun getConfigurations(project: Project, debug: Boolean = false): FileCollection {
+		val detektConfigurations = setOf(CONFIGURATION_DETEKT_PLUGIN, CONFIGURATION_DETEKT)
+		val configurations = project.configurations.filter { detektConfigurations.contains(it.name) }
+
+		val files = project.files(configurations)
+		if (debug) files.forEach { println(it) }
+		return files
+	}
 }
+
+private const val DETEKT_MAIN = "io.gitlab.arturbosch.detekt.cli.Main"

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
@@ -1,7 +1,7 @@
 package io.gitlab.arturbosch.detekt.invoke
 
 import io.gitlab.arturbosch.detekt.CONFIGURATION_DETEKT
-import io.gitlab.arturbosch.detekt.CONFIGURATION_DETEKT_PLUGIN
+import io.gitlab.arturbosch.detekt.CONFIGURATION_DETEKT_INTERNAL
 import org.gradle.api.Project
 import org.gradle.api.file.FileCollection
 
@@ -21,7 +21,7 @@ object DetektInvoker {
 	}
 
 	private fun getConfigurations(project: Project, debug: Boolean = false): FileCollection {
-		val detektConfigurations = setOf(CONFIGURATION_DETEKT_PLUGIN, CONFIGURATION_DETEKT)
+		val detektConfigurations = setOf(CONFIGURATION_DETEKT_INTERNAL, CONFIGURATION_DETEKT)
 		val configurations = project.configurations.filter { detektConfigurations.contains(it.name) }
 
 		val files = project.files(configurations)

--- a/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskKotlinDslTest.kt
+++ b/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskKotlinDslTest.kt
@@ -203,6 +203,7 @@ internal class DetektTaskKotlinDslTest : Spek({
 			assertThat(File(rootDir, "build/xml/detekt.xml")).exists()
 			assertThat(File(rootDir, "build/html/detekt.html")).exists()
 		}
+
 		it("can configure reports base directory") {
 			val detektConfig = """
 					|detekt {
@@ -227,6 +228,7 @@ internal class DetektTaskKotlinDslTest : Spek({
 			assertThat(File(rootDir, "build/my-reports/detekt.xml")).exists()
 			assertThat(File(rootDir, "build/my-reports/detekt.html")).exists()
 		}
+
 		it("can configure the input directory") {
 			val customSourceLocation = "gensrc/kotlin"
 			val detektConfig = """
@@ -249,6 +251,32 @@ internal class DetektTaskKotlinDslTest : Spek({
 
 			assertThat(result.output).contains("number of classes: 1")
 			assertThat(result.output).contains("--input, /private$rootDir/$customSourceLocation")
+			assertThat(result.task(":check")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+		}
+
+		it("can add an additional detekt dependency") {
+			val customSourceLocation = "gensrc/kotlin"
+			val detektConfig = """
+					|detekt {
+					| debug = true
+					|}
+					|
+					|dependencies {
+					| detekt("io.gitlab.arturbosch.detekt:detekt-formatting:${VERSION_UNDER_TEST}")
+					|}
+				"""
+
+			writeFiles(rootDir, detektConfig, customSourceLocation)
+			writeConfig(rootDir)
+			writeBaseline(rootDir)
+
+			// Using a custom "project-cache-dir" to avoid a Gradle error on Windows
+			val result = GradleRunner.create()
+					.withProjectDir(rootDir)
+					.withArguments("--project-cache-dir", createTempDir(prefix = "cache").absolutePath, "check", "--stacktrace", "--info")
+					.withPluginClasspath()
+					.build()
+
 			assertThat(result.task(":check")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
 		}
 	}

--- a/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskKotlinDslTest.kt
+++ b/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskKotlinDslTest.kt
@@ -254,21 +254,15 @@ internal class DetektTaskKotlinDslTest : Spek({
 			assertThat(result.task(":check")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
 		}
 
-		it("can add an additional detekt dependency") {
-			val customSourceLocation = "gensrc/kotlin"
+		it("can be used without configuration") {
 			val detektConfig = """
-					|detekt {
-					| debug = true
-					|}
-					|
 					|dependencies {
 					| detekt("io.gitlab.arturbosch.detekt:detekt-formatting:${VERSION_UNDER_TEST}")
 					|}
 				"""
 
-			writeFiles(rootDir, detektConfig, customSourceLocation)
+			writeFiles(rootDir, detektConfig)
 			writeConfig(rootDir)
-			writeBaseline(rootDir)
 
 			// Using a custom "project-cache-dir" to avoid a Gradle error on Windows
 			val result = GradleRunner.create()
@@ -277,6 +271,8 @@ internal class DetektTaskKotlinDslTest : Spek({
 					.withPluginClasspath()
 					.build()
 
+			assertThat(result.output).contains("number of classes: 1")
+			assertThat(result.output).contains("Ruleset: comments")
 			assertThat(result.task(":check")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
 		}
 	}

